homeassistant/components/loqed/coordinator.py

```diff
@@ -114,7 +115,9 @@ async def ensure_webhooks(self) -> None:
         webhook.async_register(
             self.hass, DOMAIN, "Loqed", webhook_id, self._handle_webhook
         )
-        webhook_url = webhook.async_generate_url(self.hass, webhook_id)
+
+        webhook_path = webhook.async_generate_path(webhook_id)
+        webhook_url = f"{get_url(self.hass)}{webhook_path}"
```

Comment:
Should this use `webhook.async_generate_url()` instead?

Reply:
That doesn't seem to work in some cases: https://github.com/mikewoudenberg/homeassistant-loqed/issues/10 . That's why I changed it to this more generic approach. Is this not correct?

Reply:
According to the user they got a NoURlAvailable on the `async_generate_url` call. After I switched them to the `get_url` call they reported it worked.

Reply:
Yes they are indeed using Home Assistant cloud / Nabu Casa. What approach should I take to also support that? The hardware on which I'm trying to register the webhook only supports connecting to valid https connections (so certificate matching the hostname) or plain http connections. I'm trying to find a similar example in the codebase but I'm getting lost in all the available components.

Reply:
`netatmo` is also an integration that uses the cloud webhooks you could look at. (`homeassistant.components.netatmo.register_webhook`) or search the codebase for `async_create_cloudhook`.

Please note, that webhook IDs from the cloud side should be stored and re-used.

Reply:
There is  lot going on in the Netatmo integration. I think the essence of it is checking whether the system has an active subscription and in that case registering a cloud hook instead of a regular webhook.  Do I also need to duplicate the cloud connection state listener? (so when the cloud connection ends to retrigger a webhook registration so it will register a normal webhook).

Reply:
Let's start making it just work with an active subscription. You can always make it fancier after that.

Reply:
Added (rudimentary) support for cloud hooks as proposed. Also added a testcase for it.